File: java/src/CFA.java
/**
 * @author egaeus
 * @mail sebegaeusprogram@gmail.com
 * @veredict Accepted
 * @url https://codeforces.com/problemset/problem/1353/A
 * @category greedy
 * @date 14/04/2020
 **/

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

import static java.lang.Integer.parseInt;

public class CFA {
    public static void main(String args[]) throws Throwable {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int T = parseInt(in.readLine());
        for (int t = 0; t < T; t++) {
            StringTokenizer st = new StringTokenizer(in.readLine());
            long N = parseInt(st.nextToken()), M = parseInt(st.nextToken());
            if (N == 1) System.out.println(0);
            else if(N==2) System.out.println(M);
            else {
                System.out.println(2*M);
            }
        }
    }
}

File: kotlin/src/CFA.kt
/**
 * @author egaeus
 * @mail sebegaeusprogram@gmail.com
 * @veredict Not sended
 * @url <https://codeforces.com/problemset/problem/>
 * @category ?
 * @date 29/05/2020
 **/

private fun readLn() = readLine()!! // string line
private fun readInt() = readLn().toInt() // single int
private fun readLong() = readLn().toLong() // single long
private fun readDouble() = readLn().toDouble() // single double
private fun readStrings() = readLn().split(" ") // list of strings
private fun readInts() = readStrings().map { it.toInt() } // list of ints
private fun readLongs() = readStrings().map { it.toLong() } // list of longs
private fun readDoubles() = readStrings().map { it.toDouble() } // list of doubles
private fun readString() = readLn() // single string

fun main() {
    var T = readInt();
    repeat(T) {
        var (N, K) = readInts()
        var N1 = N / (1 + K + K * K + K * K * K)
        var N2 = K * N1
        var N3 = K * N2
        var N4 = K * N3
        println("$N1 $N2 $N3 $N4")
    }
}